src/exponential_utils.jl

```diff
@@ -521,3 +537,25 @@ function phiv_timestep!(u::Vector{T}, t::Float64, A, B::Matrix{T}; tau::Float64=
 
   return u
 end
+function _phiv_timestep_adapt(m, tau, epsilon, m_old, tau_old, epsilon_old, q, kappa, gamma, omega)
+  # Compute new m and tau (Algorithm 4)
+  if tau_old > tau
+    q = log(tau/tau_old) / log(epsilon/epsilon_old) - 1
+  end # else keep q the same
+  tau_new = tau * (gamma / omega)^(1/(q + 1))
+  if m_old < m
+    kappa = (epsilon/epsilon_old)^(1/(m_old - m))
+  end # else keep kappa the same
+  m_new = m + log(omega / gamma) / log(kappa)
+  # Compare costs of using new m vs new tau
+  cost_tau = 0 # TODO
```

Comment:
integer?

Reply:
The estimation part here uses theoretical results instead of measuring actual flops. ((23) in the phipm paper).

Reply:
and the estimation is with integer values? I was just noting that this might be a place which would cause a further type-instability.

Reply:
Yes. Basically it uses the dimension `n` and `m` and count the number of `A_mul_B!`/`dot`/`axpy!` operations.

Reply:
I see